packages/contracts/test/contracts/OVM/accounts/OVM_ECDSAContractAccount.spec.ts

```diff
@@ -190,4 +190,23 @@ describe('OVM_ECDSAContractAccount', () => {
       )
     })
   })
+  describe('isValidSignature()', () => {
+    // NOTE: There is no good way to unit test verifying a valid signature
+    // An integration test exists testing this instead
```

Comment:
This works in a way of testing a valid sig.
```
it(`should return 0 for an invalid signature`, async () => {
      const messageHash = utils.hashMessage('0x42')
      const signature = await wallet.signMessage('0x42')
      const bytes = await OVM_ECDSAContractAccount.connect(wallet).isValidSignature(
        messageHash,
        signature
      )
      expect(bytes).to.equal('0x1626ba7e') // 0x1626ba7e = bytes4(keccak256("isValidSignature(bytes32,bytes)")
    })
```

Reply:
hmm I don't think this would work. `isValidSignature` checks that the signature came from `address(this)` - In unit tests I couldn't think of a way to have `OVM_ECDSAContractAccount.address` be equal to `wallet.address` - The solution in other tests has been to use `Lib_ExecutionManagerWrapper.ovmADDRESS()` and then `smock` it, but that won't work since `isValidSignature` must be a `view` function

Reply:
Erm where is address(this) reference? I only see `msg.sender` https://github.com/ethereum-optimism/optimism/pull/1052/files#diff-23c9a2eecf498d061afa1ef08165b257f94de6a06125078d3713ca8d60a54fbaR66

Reply:
I actually tested that locally and it does work. What am I missing?

Reply:
adding an integration test to catch calling `isValidSignature` from a different wallet and ensuring it still works